root_numpy/_utils.py

```diff
@@ -145,6 +150,10 @@ def stretch(arr, fields=None):
             # Scalar field
             ret[field] = np.repeat(arr[field], len_array)
 
+    if return_indices:
+        idx = np.concatenate(list(map(np.arange, len_array)))
```

Comment:
Hmm... I suppose the iter version would be faster for typical cases I've been exposed to. Probably best to write a helper function in Cython. Let's anyway merge this in as it is and I'll open an issue to translate this line into a Cythonized loop.